### PR TITLE
Fix Revoke All button to update the UI correctly

### PR DESCRIPTION
**Pull request by shubhi1092**

### CODE REVIEW

File: app/views/courses/TeacherClassView.coffee [hunk 1]
```diff
@@ -702,8 +702,15 @@ module.exports = class TeacherClassView extends RootView
...
-          success: =>
-            student.unset('coursePrepaid')
+          # The for loop completes before the success callback for the first student executes.
+          # So, the `student` will be the last student when the callback executes.
+          # Therefore, using a self calling anonymous function for the success callback
+          # to retain the student data for each iteration.
+          # Reference: https://www.pluralsight.com/guides/javascript-callbacks-variable-scope-problem
+          success: (() ->
+            st = student
+            return => st.unset('coursePrepaid')
```
Comment (AndrewJakubowicz) [inline app/views/courses/TeacherClassView.coffee]: Is a `fat arrow` required?

Comment (AndrewJakubowicz) [reply app/views/courses/TeacherClassView.coffee]: Also just for fun. You could pass the student into the anonymous function like so:

```
success: ((st) -> () -> st.unset('coursePrepaid'))(student)
```

Comment (AndrewJakubowicz) [reply app/views/courses/TeacherClassView.coffee]: Which could then be simplified to:

```
success: (st -> -> st.unset('coursePrepaid'))(student)
```

Comment (shubhi1092) [reply app/views/courses/TeacherClassView.coffee]: Yeah, interesting. :)
Not sure about the `fat arrow` question though. Do you think a `thin arrow` would also be fine?

**Pull request merged**